opnsense/resource_firewall_alias.go

```diff
@@ -0,0 +1,317 @@
+package opnsense
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/kradalby/opnsense-go/opnsense"
+	"github.com/satori/go.uuid"
+	"log"
+	"strings"
+)
+
+func resourceFirewallAlias() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceFirewallAliasCreate,
+		Read:   resourceFirewallAliasRead,
+		Update: resourceFirewallAliasUpdate,
+		Delete: resourceFirewallAliasDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"parent": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "UIID of other alias who will contain this alias (nested)",
+				Optional:    true,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "State of the alias",
+				Optional:    true,
+				Default:     true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Name of the alias",
+				Required:    true,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Description:  "Type of the alias",
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"host", "network", "port", "url"}, false),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of the alias",
+				Optional:    true,
+				Default:     "",
+			},
+			"content": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The content of this alias (IP, Cidr, url, ...)",
+
+				Optional: true,
+			},
+			// TODO add other fields (like proto)
+		},
+	}
+}
+
+func resourceFirewallAliasRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] Getting OPNsense client from meta")
+	c := meta.(*opnsense.Client)
+
+	log.Printf("[TRACE] Converting ID to UUID")
+	uuid, err := uuid.FromString(d.Id())
+	if err != nil {
+		log.Printf("[ERROR]resourceFirewallAliasRead -  Failed to parse ID")
+		return err
+	}
+
+	log.Printf("[TRACE] Fetching alias configuration from OPNsense")
+	alias, err := c.AliasGet(uuid)
+	if err != nil {
+		// temporary fix for the internal error API when we try to get an unreferenced UIID
+		if err.Error() == "Internal Error status code received" {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("ERROR: \n%#v", err)
```

Comment:
This error message looks like its for debug?, if not, could it be made similar to the rest of the log messages for consistency? Also maybe not two in a row.